core/src/main/codegen/templates/Parser.jj

```diff
@@ -1307,6 +1308,10 @@ SqlSelect SqlSelect() :
     |   <ALL> {
             keywords.add(SqlSelectKeyword.ALL.symbol(getPos()));
         }
+<#if parser.includeTopN>
+    |
+        topN = SqlSelectTopN(getPos())
```

Comment:
So does this mean that the TOP N clause is mutually exclusive with the DISTINCT keyword?

Reply:
Yes, this is what the syntax diagram says the behavior should be.